compiler/rustc_hir_analysis/src/impl_wf_check.rs

```diff
@@ -181,6 +181,15 @@ fn report_unused_parameter(tcx: TyCtxt<'_>, span: Span, kind: &str, name: Symbol
         name
     );
     err.span_label(span, format!("unconstrained {} parameter", kind));
+    err.span_suggestion(
+        span,
+        format!(
+            "Either remove the type parameter '{}' or make use of it, for example ` for S<{}>`.",
```

Comment:
we should not just suggest `S<T>`. Either load the trait or type name and append the generics to that or write prosa explaining what's going on without naming the trait or type.